display/renderawt/src/main/java/org/openscience/cdk/renderer/visitor/AWTDrawVisitor.java

```diff
@@ -206,7 +198,45 @@ private void visit(LineElement line) {
         graphics.setStroke(savedStroke);
     }
 
-    private void visit(OvalElement oval) {
+	/**
+	 * 
+	 * Create and set the stroke, optionally storing 2.25, 2.5, 2.75 etc to 
+	 * separate keys in strokeCache.
+	 * 
+	 * Strokes are keyed as Integer.valueOf(int) (4*width)), negated for default style (square, miter)
+	 * 
+	 * Optionally set the stroke to rounded or square.
+	 * 
+	 * 
+	 * @param dwidth calcuated width
+	 * @param rounded true for rounded stroke; false for square
+	 * @return current stroke
+	 * @author Bob Hanson
+	 */
+    private Stroke setStroke(double dwidth, boolean rounded) {
+    	float width = (float) dwidth;
+        if (width < minStroke) 
+        	width = minStroke;
+        Integer key = (strokeCache ? Integer.valueOf((int) (width * 4) * (rounded ? 1 : -1)) : null);
+        Stroke savedStroke = this.graphics.getStroke();
+        BasicStroke stroke = (strokeCache ? strokeMap.get(key) : null);
+        if (stroke == null) {
+        	// TODO: John, please check.
+        	// There was a second bug here in that both straight and 
+        	// rounded were being stored with the same key. 
```

Comment:
Are you sure? In the current version I only see CAP_ROUND being cached, which makes sense as it is only the main bond lines we want fractional coords.